internal/crypto/src/time_stamp/response.rs

```diff
@@ -0,0 +1,192 @@
+            .map_err(|e| TimeStampError::DecodeError(e.to_string()))?,
+        ))
+    }
+
+    /// Return `true` if the request was successful.
+    pub fn is_success(&self) -> bool {
+        matches!(
+            self.0.status.status,
+            PkiStatus::Granted | PkiStatus::GrantedWithMods
+        )
+    }
+
+    /// TO REVIEW: Does this need to be public?
+    pub fn signed_data(&self) -> Result<Option<SignedData>, TimeStampError> {
+        if let Some(token) = &self.0.time_stamp_token {
+            if token.content_type == OID_ID_SIGNED_DATA {
+                Ok(Some(
+                    token
+                        .content
+                        .clone()
+                        .decode(SignedData::take_from)
+                        .map_err(|e| TimeStampError::DecodeError(e.to_string()))?,
+                ))
+            } else {
+                Err(TimeStampError::DecodeError(
+                    "Invalid OID for signed data".to_string(),
+                ))
+            }
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// TO REVIEW: Does this need to be public?
+    pub fn tst_info(&self) -> Result<Option<TstInfo>, TimeStampError> {
+        if let Some(signed_data) = self.signed_data()? {
+            if signed_data.content_info.content_type == OID_CONTENT_TYPE_TST_INFO {
+                if let Some(content) = signed_data.content_info.content {
+                    Ok(Some(
+                        Constructed::decode(content.to_bytes(), bcder::Mode::Der, |cons| {
+                            TstInfo::take_from(cons)
+                        })
+                        .map_err(|e| TimeStampError::DecodeError(e.to_string()))?,
+                    ))
+                } else {
+                    Ok(None)
+                }
+            } else {
+                Ok(None)
+            }
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Convert this time stamp response to a time stamp token.
+    pub fn time_stamp_token(&self) -> Option<Vec<u8>> {
+        let tst = self.0.time_stamp_token.as_ref()?;
+
+        let arcs: Result<Vec<u32>, TimeStampError> = tst
+            .content_type
+            .iter()
+            .map(|v| {
+                v.to_u32().ok_or(TimeStampError::DecodeError(
+                    "no content type specified".to_string(),
+                ))
+            })
+            .collect();
+
+        let arcs = arcs.ok()?;
+
+        let ci = ContentInfo {
+            content_type: rasn::types::ObjectIdentifier::new(arcs)?,
+            content: rasn::types::Any::new(tst.content.as_bytes().to_vec()),
+        };
+
+        rasn::der::encode(&ci).ok()
+    }
+}
+
+#[derive(AsnType, Clone, Debug, Decode, Encode, PartialEq, Eq, PartialOrd, Ord, Hash)]
+struct ContentInfo {
+    content_type: rasn::types::ObjectIdentifier,
+
+    #[rasn(tag(explicit(0)))]
+    content: rasn::types::Any,
+}
+
+/// TO REVIEW: Does this need to be public after refactoring?
```

Comment:
This is used by verify_timestamp functions.

Reply:
Still used, but downgraded to `pub(crate)`